assets/src/edit-story/panels/size/useSize.js

```diff
@@ -0,0 +1,59 @@
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback, useState, useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+
+function useSize( width, height ) {
+	const [ lockRatio, setLockRatio ] = useState( true );
+	const [ state, setState ] = useState( { width, height } );
+
+	useEffect( () => {
+		setState( { width, height } );
+	}, [ width, height, setState ] );
+
+	const handleToggleLockRatio = useCallback( () => {
+		setLockRatio( ( val ) => ! val );
+	}, [ setLockRatio ] );
+
+	const handleWidth = useCallback( ( value ) => {
+		const ratio = width / height;
+		const newWidth = isNaN( value ) || value === '' ? '' : parseFloat( value );
+		setState( {
+			...state,
+			width: newWidth,
+			height: typeof newWidth === 'number' && lockRatio ? newWidth / ratio : height,
+		} );
+	}, [ width, height, state, setState, lockRatio ] );
+
+	const handleHeight = useCallback( ( value ) => {
+		const ratio = width / height;
+		const newHeight = isNaN( value ) || value === '' ? '' : parseFloat( value );
+		setState( {
+			...state,
+			height: newHeight,
+			width: typeof newHeight === 'number' && lockRatio ? newHeight * ratio : width,
+		} );
+	}, [ width, height, state, setState, lockRatio ] );
+
+	const handleSubmit = useCallback( ( onSetProperties ) => ( event ) => {
+		onSetProperties( state );
+		event.preventDefault();
+	}, [ state ]	);
+
+	return {
+		lockRatio,
+		state,
+		setState,
```

Comment:
Do you still need `setState` outside? The true value of custom hooks is limiting the exposure to only the variables and functions needed.

Reply:
Actually we don't need it